src/calculator/index.ts

```diff
@@ -325,14 +334,14 @@ export default class Calculator {
 
       const voteAmountInUsd = convertTokenToFiat({
         tokenAmount: vote.amount,
-        tokenDecimals: tokenPrice.decimals,
```

Comment:
Your suggested code gives me the same results for a polygon round with usdc, but I am unsure of the deeper implications. I think a lot of this code warrants a little cleanup, which we are already doing with Max.

Reply:
I would rather merge what works, and then we can further improve the code we have here.

Reply:
this PR is using the round token to convert the voting token to fiat, it is wrong

Reply:
The bug here is that we're using the hardcoded 18 to convert from fiat to round token, the variable names maybe make this confusing, the process is as follows:

1. We take the voting token, we convert it to fiat (USD) using the decimals of the voting token
2. We take the fiat (USD) and convert it to the round token, using the decimals of the round token

The voting token and the round token can have different decimals

We always truncate the precision of the USD price to 8 decimal places, this is hardcoded

Essentially this bug is probably happening because we have a round denominated in USDC

Reply:
ah okay, i see what you mean, fixed according to your suggestions.